django_meili/querysets.py

```diff
@@ -259,5 +259,5 @@ def search(self, q: str = ""):
                 "attributesToSearchOn": self.__attributes_to_search_on,
             },
         )
-        hits = results["hits"]
-        return self.model.objects.filter(pk__in=[hit["id"] for hit in hits])
+        id_field = getattr(self.model.MeiliMeta, "primary_key", "id")
+        return self.model.objects.filter(pk__in=[hit.get(id_field) for hit in results.get("hits", [])])
```

Comment:
If we're going to have None in the list, let's filter those out.

Reply:
None? sorry not sure what i missed, none should never be in list?

Reply:
The default value for `dict.get` is None. If we have the ability for None to exist we should filter, otherwise let's just use direct access. I'd rather it crash based on malformed data than silently fail.

Reply:
I agree, then let's just use direct key access. So instead of
```python
return self.model.objects.filter(pk__in=[hit.get(id_field) for hit in results.get("hits", [])])
```
we do
```python
return self.model.objects.filter(pk__in=[hit[id_field] for hit in results.get("hits", [])])
```

Reply:
i guess its a matter of taste.. but i really tend to avoid using square brackets notation as its a recipe for unexpected IndexError

Reply:
That's my point. I want it to fail if its invalid. I want it to throw something if there's an issue here. I usually avoid square brackets as well, but in this case I think a silent failure is just going to confuse an end user.